docs/src/main/java/io/quarkus/docs/generation/AllConfigGenerator.java

```diff
@@ -0,0 +1,149 @@
+package io.quarkus.docs.generation;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.resolution.ArtifactRequest;
+import org.eclipse.aether.resolution.ArtifactResult;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.quarkus.annotation.processor.generate_doc.ConfigDocItem;
+import io.quarkus.annotation.processor.generate_doc.ConfigDocItemScanner;
+import io.quarkus.annotation.processor.generate_doc.ConfigDocSection;
+import io.quarkus.annotation.processor.generate_doc.ConfigDocWriter;
+import io.quarkus.bootstrap.resolver.AppModelResolverException;
+import io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver;
+import io.quarkus.docs.generation.ExtensionJson.Extension;
+
+public class AllConfigGenerator {
+    public static void main(String[] args)
+            throws AppModelResolverException, JsonParseException, JsonMappingException, IOException {
+        MavenArtifactResolver resolver = MavenArtifactResolver.builder().build();
+        // This is where we produce the entire list of extensions
+        String jsonPath = "devtools/core-extensions-json/target/extensions.json";
```

Comment:
Can we build the `docs` module without running a full build Quarkus build?

Reply:
No, because the extension list is generated by the build.

Reply:
Okay, maybe we can safely silently ignore generating the `all-config.adoc` file if we do not have the `extensions.json` file. There are cases where someone may want to quickly build the `docs` module to check the generated output following documentation change. We should not require a full build for this case.

Reply:
I agree with @machi1990 . There's a good chance it will be very painful for doc authors otherwise.

Reply:
OK, lemme handle that.

Reply:
Done